org.eclipse.xtext/src/org/eclipse/xtext/formatting2/debug/TextRegionAccessToString.java

```diff
@@ -349,14 +361,17 @@ protected String toString(EObject ele) {
 			return grammarToString.apply((AbstractElement) ele);
 		if (ele instanceof AbstractRule)
 			return ele.eClass().getName() + "'" + ((AbstractRule) ele).getName() + "'";
+		if (ele == null) {
```

Comment:
@cdietrich This is the NPE fix

Reply:
@tivervac  any hint on how to get an IComment without grammar element?

Reply:
In our case it was a syntax error. We have a hidden terminal rule that looks somewhat like this
```
terminal MACRO: '`' ('a'..'z');
```

When you then have code that only has the backtick but no body we would have an IComment without grammar element because the node behind the IComment was a `HiddenLeafNodeWithSyntaxError`.

```
package p;
  `
endpackage
```